src/Cache/Adapter/Sharding.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+namespace Utopia\Cache\Adapter;
+
+use Utopia\Cache\Adapter;
+
+class Sharding implements Adapter
+{
+    /**
+     * @var Adapter[] 
+     */
+    protected $adapters;
+
+    /**
+     * Sharding constructor.
+     * 
+     * Allows to shard cache across multiple adapters in a consistent way.
+     * 
+     * Please Note:
+     *  When adding a new adapter to the pool, cached will gradually
+     *  get re-distributed to fill the new adapter, this might casue a siegnificant drop
+     *  in hit-rate for a short period of time.
+     * 
+     * @param Adapter[] $adapters
+     */
+    public function __construct(array $adapters)
+    {
+        $this->adapters = $adapters;
+    }
```

Comment:
we might wanna check here if the array is not empty and throw an exception otherwise 👍🏻